### PR TITLE
Ft/create comments

### LINKED ISSUE #27: Create a Comment
As a commenter, I would like to be able to add a Comment to a Post so that I can take part in the discussion about a Post.

**Given** the user is viewing the Details of a Post
**When** they select the `Add Comment` menu option
**Then** they should be directed to a page with a form for creating a new Comment

**Given** the user has entered the relevant information for a Comment
**When** they click the `Save` button
**Then** the Comment should be saved to the database
**And** the creation datetime should be automatically set to the current date and time
**And** the user should be redirected to the new Comments list page for the related Post  

The "relevant information" for a Comment is

* Subject
* Content

### PR DESCRIPTION
**Pull request by LucasBress0n**
Purpose: Allow users to create comments on a post

Files Changed:
Controllers/CommentController.cs
client/src/components/ApplicationViews.jsx
client/src/components/CommentForm.jsx
client/src/components/PostDetails.jsx
client/src/managers/comment.js

 To Test:
As a User or Admin head to a post details, hit the add comment button and fillout your subject(small input) and content(large textarea) then hit surbmurterionation to save.

Closes: #27

### CODE REVIEW

File: Controllers/CommentController.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    public CommentController(TabloidDbContext context)
+    {
+        _db = context;
+    }
+
+
+    [HttpPost]
+    [Authorize]
+    public IActionResult AddNewComment(Comment newComment)
```
Comment (Lincoln-WIlcox) [inline Controllers/CommentController.cs]: Make a DTO so the `CreationDate` and `Id` properties are not exposed to the client.

File: client/src/components/CommentForm.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    });
+  };
+
+  return (
+    <main>
+      <div>
+        <form onSubmit={handleSave}>
+          <input
+            required
+            defaultValue={commentObject.subject}
```
Comment (Lincoln-WIlcox) [inline client/src/components/CommentForm.jsx]: Is there a reason this is `defaultValue` and not `value`? Seems like this should be a controlled element.

File: client/src/components/CommentForm.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            onChange={(e) => {
+              const copy = { ...commentObject };
+              copy.subject = e.target.value;
+              setCommentObject(copy);
+            }}
+          />
+          <textarea
+            required
+            defaultValue={commentObject.content}
```
Comment (Lincoln-WIlcox) [inline client/src/components/CommentForm.jsx]: Here too

**Review by rubygkendrick — APPROVED** [review]
this works

**Review by Lincoln-WIlcox — CHANGES_REQUESTED** [review]

File: client/src/components/CommentForm.jsx (hunk 2 above)
Comment (LucasBress0n) [reply client/src/components/CommentForm.jsx]: It's considered an uncontrolled element when using value, probably due to subject not existing till you input something into the onchange.

File: Controllers/CommentController.cs (hunk 1 above)
Comment (LucasBress0n) [reply Controllers/CommentController.cs]: I'm lost on this part, the Id and CreationDate are never exposed to the front-end as this is posting a new Comment.

File: client/src/components/CommentForm.jsx (hunk 2 above)
Comment (Lincoln-WIlcox) [reply client/src/components/CommentForm.jsx]: It is considered a controlled element when using value. I assume you're talking about the error you get when you change the field. Initially, value will be undefined as `commentObject.subject` is undefined. When value is undefined, react treats it as uncontrolled. When it becomes defined, it switches to being controlled, which react doesn't like for some reason. You can resolve this error by setting an initial value for `commentObject.subject` (probably an empty string).

File: Controllers/CommentController.cs (hunk 1 above)
Comment (Lincoln-WIlcox) [reply Controllers/CommentController.cs]: There aren't fields for those in the form, but if this was being hosted publicly anyone could call this endpoint and fill those fields. You can read more regarding using DTOs to expose data in [this](https://ardalis.com/web-api-dto-considerations/) article under the Create header.

**Review by Lincoln-WIlcox — APPROVED** [review]

**Pull request merged**